core/src/avm2/verify.rs

```diff
@@ -0,0 +1,1193 @@
+                                Op::NewFunction { .. } => {
+                                    if GcCell::ptr_eq(
+                                        class,
+                                        activation
+                                            .avm2()
+                                            .classes()
+                                            .function
+                                            .inner_class_definition(),
+                                    ) {
+                                        previous_op = Some(op.clone());
+                                        *op = Op::Nop;
+                                        continue;
+                                    }
+                                }
+                                Op::Coerce {
+                                    index: previous_name_index,
+                                } => {
+                                    if name_index.as_u30() == previous_name_index.as_u30() {
+                                        previous_op = Some(op.clone());
+                                        *op = Op::Nop;
+                                        continue;
+                                    }
+                                }
+                                _ => {}
+                            }
+                        }
+                    }
+                    _ => {}
+                }
+            }
+        }
+
+        previous_op = Some(op.clone());
+    }
+}
+
+fn ops_can_throw_error(ops: &[AbcOp]) -> bool {
+    for op in ops {
+        match op {
+            AbcOp::Bkpt
+            | AbcOp::BkptLine { .. }
+            | AbcOp::Timestamp
+            | AbcOp::PushByte { .. }
+            | AbcOp::PushDouble { .. }
+            | AbcOp::PushFalse
+            | AbcOp::PushInt { .. }
+            | AbcOp::PushNamespace { .. }
+            | AbcOp::PushNaN
+            | AbcOp::PushNull
+            | AbcOp::PushShort { .. }
+            | AbcOp::PushString { .. }
+            | AbcOp::PushTrue
+            | AbcOp::PushUint { .. }
+            | AbcOp::PushUndefined
+            | AbcOp::Dup
+            | AbcOp::Swap
+            | AbcOp::Pop
+            | AbcOp::TypeOf
+            | AbcOp::GetGlobalScope
+            | AbcOp::GetScopeObject { .. }
+            | AbcOp::GetOuterScope { .. }
+            | AbcOp::GetGlobalSlot { .. }
+            | AbcOp::GetLocal { .. }
+            | AbcOp::SetLocal { .. }
+            | AbcOp::Kill { .. }
+            | AbcOp::Label
+            | AbcOp::Jump { .. }
+            | AbcOp::IfTrue { .. }
+            | AbcOp::IfFalse { .. }
+            | AbcOp::IfStrictEq { .. }
+            | AbcOp::IfStrictNe { .. }
+            | AbcOp::LookupSwitch { .. }
+            | AbcOp::Nop
+            | AbcOp::Not
+            | AbcOp::PopScope
+            | AbcOp::ReturnVoid => {}
+            _ => return true,
+        }
+    }
+
+    false
+}
+
+fn pool_int<'gc>(
```

Comment:
can we move these to `script.rs`? It already has a bunch of similar functions, so it'd be more consistent (and decrease verify.rs size a bit too).
Also, any chance this could be unified with `abc_int()` (and friends)? As they are pretty much duplicates, sans `0` treatment. (this'd also probably make function with invalid default values throw a correct VerifyError "for free"). Can be left for a follow-up though.

Reply:
I'm going to leave this for a follow-up to also fix more issues with namespace and string pooling